server/test/unit/server/managers/content/test_upload.py

```diff
@@ -223,6 +223,17 @@ def test_import_uploaded_unit_importer_error_reraise_pulp_exception(self, mock_r
         self.assertRaises(InvalidValue, self.upload_manager.import_uploaded_unit, 'repo-u',
                           'mock-type', {}, {}, upload_id)
 
+    @mock.patch('pulp.server.controllers.importer.model.Repository.objects')
+    def test_import_uploaded_unit_success_flag_false(self, mock_repo_qs):
+        """Test that exception is raised if upload report indicates failure."""
+        importer_controller.set_importer('repo-u', 'mock-importer', {})
+        importer_return_report = {'success_flag': False, 'summary': '', 'details': {}}
+        mock_plugins.MOCK_IMPORTER.upload_unit.side_effect = None
+        mock_plugins.MOCK_IMPORTER.upload_unit.return_value = importer_return_report
+        upload_id = self.upload_manager.initialize_upload()
+        self.assertRaises(PulpCodedException, self.upload_manager.import_uploaded_unit, 'repo-u',
+                          'mock-type', {}, {}, upload_id)
```

Comment:
I definitely prefer with statement, but all the tests are rather old(?) in this file and they are written that way.
So I thought that maybe it would be easier to rewrite all these unittests at once someday and not to add one new-style test to it. What do you think?